pkg/flowaggregator/flowaggregator.go

```diff
@@ -166,21 +169,52 @@ func NewFlowAggregator(externalFlowCollectorAddr string, externalFlowCollectorPr
 		nil,
 		0,
 		registry,
+		flowAggregatorDNSName,
+		k8sClient,
 	}
 	return fa
 }
 
-func genObservationID() (uint32, error) {
+func (fa *flowAggregator) genObservationID() (uint32, error) {
 	// TODO: Change to use cluster UUID to generate observation ID once it's available
 	h := fnv.New32()
-	h.Write([]byte(flowAggregatorDNSName))
+	h.Write([]byte(fa.flowAggregatorDNSName))
 	return h.Sum32(), nil
 }
 
 func (fa *flowAggregator) InitCollectingProcess() error {
 	var err error
 	var cpInput collector.CollectorInput
-	if fa.aggregatorTransportProtocol == AggregatorTransportProtocolTCP {
+	if fa.aggregatorTransportProtocol == AggregatorTransportProtocolTLS {
+		parentCert, privateKey, caCert, err := generateCACertKey()
+		if err != nil {
+			return fmt.Errorf("error when generating CA certificate: %v", err)
+		}
+		// TODO: support IP input for creating server certificate
```

Comment:
do you think any change will be required here. we can just use an IP address as the value for `flowAggregatorDNSName` in flow-aggregator.conf? At least I don't think any change is required in the Antrea code, but maybe in the go-ipfix code?

Reply:
Only a little code: check whether the input is DNS name or IP address, then assign them to the server certificate. I didn't see any changes needed in go-ipfix. We can include this change in this PR. What do you think?

Reply:
I think that regardless of whether it is an actual DNS name of an IP address, the value will go into the `DNSNames` field when you build the certificate. What kind of special handling do you think is needed?

Reply:
For IP address, it will go into IPAddresses field. I can have a quick change on that.

Reply:
I see, good point. I forgot there was a dedicated field. Yes calling `net.ParseIP` as part of this PR and using the correct field when building the certificate would be good. Thanks!